### PR TITLE
Search page test base branch

### PR DESCRIPTION
**Pull request by dualcnhq**

### CODE REVIEW

File: .github/workflows/call-run-tests.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
         run: npx playwright install --with-deps
 
       - name: Run Playwright tests
-        run: npm run gh:test
+        run: npm run test
```
Comment (iisa) [inline .github/workflows/call-run-tests.yml]: @dualcnhq I thought you mentioned that we are clearing these workflow changes from CR?

File: tests/search/search-view-mode.spec.ts [hunk 2]
```diff
@@ -1,94 +0,0 @@
...
-  await expect(infiniteScroller.locator('item-tile').first()).not.toBeVisible();
-  await expect(infiniteScroller.locator('tile-list').first()).toBeVisible();
-  await expect(infiniteScroller.locator('tile-list-compact').first()).not.toBeVisible();
-
-  // view mode - click list-compact
-  await displayStyleSelectorOptions.nth(2).click();
-  await expect(infiniteScroller).toHaveClass(/list-compact/);
-  await expect(infiniteScroller.locator('item-tile').first()).not.toBeVisible();
-  await expect(infiniteScroller.locator('tile-list').first()).not.toBeVisible();
-  await expect(infiniteScroller.locator('tile-list-compact').first()).toBeVisible();
```
Comment (iisa) [inline tests/search/search-view-mode.spec.ts:92]: Q: why did we delete some of these tests?

File: tests/search/search-test.spec.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    await searchTextContents.click();
+    await collectionSearchGo.click();
+    await expect(page).toHaveURL(/query=cats&sin=TXT/);
+
+    await searchRadioTranscripts.click();
+    await collectionSearchGo.click();
+    await expect(page).toHaveURL(/query=cats&sin=RADIO/);
+  });
+
+});
```
Comment (iisa) [inline tests/search/search-test.spec.ts]: Q: what has changed here? this seems like a reorg of the ones that were deleted?

File: tests/search/search-base.spec.ts [hunk 4]
```diff
@@ -1,29 +0,0 @@
-import { expect } from '@playwright/test';
-import { test } from '../../fixtures';
-
-test('page load - initial load', async ({ page }) => {
```
Comment (iisa) [inline tests/search/search-base.spec.ts:4]: this blob looks equivalent to https://github.com/internetarchive/archiveorg-e2e-playwright/pull/8/files#diff-1f2654a88c34cf16b56f72b9ec74be2308e0a8c0f9db95164fde6e911bb8ec9dR19 - what changed?

File: tests/search/search-facets.spec.ts [hunk 5]
```diff
@@ -1,48 +0,0 @@
...
-  const facetsHeaderContainer = page.locator('div#facets-header-container');
-  const facetsContainer = page.locator('div#facets-container');
-  const facetGroups = facetsContainer.locator('section.facet-group');
-  const headerTitles = facetsContainer.locator('h3');
-
-  expect(await leftColumn.count()).toEqual(1);
-  expect(await facetsHeaderContainer.count()).toEqual(1);
-  expect(await facetsContainer.count()).toEqual(1);
-  expect(await facetGroups.count()).toEqual(8);
-  expect(await headerTitles.count()).toEqual(8);
```
Comment (iisa) [inline tests/search/search-facets.spec.ts:29]: this looks like this https://github.com/internetarchive/archiveorg-e2e-playwright/pull/8/files#diff-1f2654a88c34cf16b56f72b9ec74be2308e0a8c0f9db95164fde6e911bb8ec9dR46 - what is difference?
looks like we lost the block below as well?

File: tests/search/search-test.spec.ts [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    await page.getByRole('button', { name: 'Toggle options Date archived' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'Date reviewed' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-reviewdate/);
+
+    await page.getByRole('button', { name: 'Toggle options Date reviewed' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'Date added' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-addeddate/);
+
+    await page.getByRole('button', { name: 'Creator' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=creator/);
```
Comment (iisa) [inline tests/search/search-test.spec.ts]: Q: do we need to test result set here? to confirm that they are truly being sorted?

File: tests/search/search-test.spec.ts [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    const searchTextContents = buttonCollapserLocator.getByText('Search text contents');
+    const searchTVCaptions = buttonCollapserLocator.getByText('Search TV news captions');
+    const searchRadioTranscripts = buttonCollapserLocator.getByText('Search radio transcripts');
+    const searchArchivedWebSites = buttonCollapserLocator.getByText('Search archived web sites');
+    
+    expect(await searchMetadata.count()).toEqual(1);
+    expect(await searchTextContents.count()).toEqual(1);
+    expect(await searchTVCaptions.count()).toEqual(1);
+    expect(await searchRadioTranscripts.count()).toEqual(1);
+    expect(await searchArchivedWebSites.count()).toEqual(1);
```
Comment (iisa) [inline tests/search/search-test.spec.ts]: these are ok tests, they check the value and existence but do not check for behaviour? how can we test them to see results working?

File: tests/search/search-test.spec.ts [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+
+    await page.getByRole('button', { name: 'Toggle options Date reviewed' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'Date added' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-addeddate/);
+
+    await page.getByRole('button', { name: 'Creator' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=creator/);
+  });
+  
+  test('collection-search-input options', async () => {
```
Comment (iisa) [inline tests/search/search-test.spec.ts]: Q: what does this describe? I'm unsure what behavior this is checking for?
suggestion: create a title that describes behavior not ui..

File: tests/search/search-test.spec.ts [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    await page.getByRole('button', { name: 'Relevance' }).click();
+
+    await page.getByText('Weekly views').first().click();
+    await expect(page).toHaveURL(/query=cats&sort=-week/);
+
+    await page.getByRole('button', { name: 'Toggle options Weekly views' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'All-time views' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-downloads/);
+
+    await page.getByRole('button', { name: 'Title' }).click();
```
Comment (iisa) [inline tests/search/search-test.spec.ts]: I think it would be good, with each click - that it checks the results set being displayed...

**Review by iisa — CHANGES_REQUESTED** [review]
a good start. I've left suggestions here and am wanting more robustness for these tests - let's make sure we test results set on every Sort/facet selection that you make.  I think that will help stabilize the outcome of what we expect.

File: .github/workflows/call-run-tests.yml (hunk 1 above)
Comment (dualcnhq) [reply .github/workflows/call-run-tests.yml]: yes, but this specific workflow is also used in the main workflow.

the `npm run gh:test` has a `--trace on` parameter which takes a lot of memory on GH web runner so switching it back for now to the usual run test command

File: tests/search/search-page-url-load.spec.ts [hunk 10]
```diff
@@ -0,0 +1,228 @@
+/* Scenario: 
+Search Page initial page load and URL
```
Comment (iisa) [inline tests/search/search-page-url-load.spec.ts:2]:
```suggestion
Search Page: User searches for "cats"
```

**Comment by dualcnhq** [comment]
linked PR to: https://github.com/internetarchive/archiveorg-e2e-playwright/pull/10

**Comment by dualcnhq** [comment]
going to merge this and create new PR for further updates

**Pull request merged**